src/noisepy/seis/asdfstore.py

```diff
@@ -60,3 +61,85 @@ def _create_station(self, ds: pyasdf.ASDFDataSet, name: str) -> Optional[Station
         except Exception as e:
             logger.warning(f"Missing StationXML for station {name}. {e}")
             return None
+
+
+class ASDFCCStore(CrossCorrelationDataStore):
+    def __init__(self, directory: str) -> None:
+        super().__init__()
+        self.directory = directory
+        Path(directory).mkdir(exist_ok=True)
+
+    def contains(
+        self, timespan: DateTimeRange, src_chan: Channel, rec_chan: Channel, parameters: ConfigParameters
+    ) -> bool:
+        key = get_channel_pair_key(src_chan, rec_chan)
+        exists = self._tmp_contains(timespan, key)
+        if exists:
+            logger.info(f"Cross-correlation {key} already exists")
+        return exists
+
+    def save_parameters(self, parameters: ConfigParameters):
+        fc_metadata = os.path.join(self.directory, "fft_cc_data.txt")
+
+        fout = open(fc_metadata, "w")
+        # WIP actually serialize this
+        fout.write(str(parameters.__dict__))
+        fout.close()
+
+    def append(
+        self,
+        timespan: DateTimeRange,
+        src_chan: Channel,
+        rec_chan: Channel,
+        params: ConfigParameters,
+        cc_params: Dict[str, Any],
+        corr: np.ndarray,
+    ):
+        filename = f"{os.path.join(self.directory, self._get_filename(timespan))}.h5"
+        with pyasdf.ASDFDataSet(filename, mpi=False) as ccf_ds:
+            # source-receiver pair
+            data_type = f"{src_chan.station}_{rec_chan.station}"
+            path = f"{src_chan.type}_{rec_chan.type}"
+            data = np.zeros(corr.shape, dtype=corr.dtype)
+            data[:] = corr[:]
+            ccf_ds.add_auxiliary_data(data=data, data_type=data_type, path=path, parameters=cc_params)
+
+        # record the progress
+        self._append_tmp(timespan, get_channel_pair_key(src_chan, rec_chan))
+
+    def is_done(self, timespan: DateTimeRange):
+        done = self._tmp_contains(timespan, DONE_MARKER)
+        if done:
+            logger.info(f"Timespan {timespan} already computed")
+        return done
+
+    def mark_done(self, timespan: DateTimeRange):
+        self._append_tmp(timespan, DONE_MARKER)
+
+    def read(self, chan1: Channel, chan2: Channel, start: datetime, end: datetime) -> np.ndarray:
+        pass
+
+    def _get_filename(self, timespan: DateTimeRange) -> str:
+        return f"{timespan.start_datetime.strftime(DATE_FORMAT)}T{timespan.end_datetime.strftime(DATE_FORMAT)}"
+
+    def _get_tmp_filename(self, timespan: DateTimeRange) -> str:
+        base_filename = os.path.join(self.directory, self._get_filename(timespan))
+        return f"{base_filename}.tmp"
+
+    def _append_tmp(self, timespan: DateTimeRange, line: str):
+        tmp_filename = self._get_tmp_filename(timespan)
+        with open(tmp_filename, "a") as ftmp:
+            ftmp.write(f"{line}\n")
+
+    def _tmp_contains(self, timespan: DateTimeRange, key: str):
```

Comment:
I agree there are potentially better ways of tracking progress. Since this is a refactoring I'm trying not to change implementation in the same PR. But yes, let's revisit.